filebeat/input/file.go

```diff
@@ -55,17 +58,38 @@ func (f *FileEvent) GetState() *FileState {
 	return state
 }
 
+// mergeJSONFields writes the JSON fields in the event map,
+// respecting the KeysUnderRoot and OverwriteKeys configuration options.
+func mergeJSONFields(f *FileEvent, event common.MapStr) {
+	if f.JSONConfig.KeysUnderRoot {
+		for k, v := range f.JSONFields {
```

Comment:
I thought about it, but as it's currently implemented it wouldn't have worked (assumes the container name is `fields`) and I didn't want to complicate that code.